### PR TITLE
[ci skip] Add GFX942_APU to Makefile

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: Makefile.kokkos [hunk 1]
```diff
@@ -1207,10 +1209,15 @@ ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX940), 1)
   tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx940\"")
   KOKKOS_INTERNAL_AMD_ARCH_FLAG := --offload-arch=gfx940
 endif
-ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX942), 1)
-  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GFX942")
-  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942\"")
-  KOKKOS_INTERNAL_AMD_ARCH_FLAG := --offload-arch=gfx942
+ifeq ($(kokkos_internal_use_arch_amd_gfx942), 1)
+  tmp := $(call kokkos_append_header,"$h""define kokkos_arch_amd_gfx942")
+  tmp := $(call kokkos_append_header,"$h""define kokkos_arch_amd_gpu \"gfx942\"")
```
Comment (masterleinad) [inline Makefile.kokkos:1214]: What happened here? Why is everything lowercase?

Comment (crtrott) [reply Makefile.kokkos:1214]: ups no idea I was just not looking far enough but somehow the copy paste in vim made it all lower case ...

Comment (crtrott) [reply Makefile.kokkos]: fixed

File: Makefile.kokkos [hunk 2]
```diff
@@ -1212,6 +1214,11 @@ ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX942), 1)
   tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942\"")
   KOKKOS_INTERNAL_AMD_ARCH_FLAG := --offload-arch=gfx942
 endif
+ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX942_APU), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GFX942_APU")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942\"")
```
Comment (JBludau) [inline Makefile.kokkos:1219]:
```suggestion
  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942_APU\"")
```
or is it referring to the base architecture?

**Review by masterleinad — APPROVED** [review]

File: Makefile.kokkos (hunk 2 above)
Comment (crtrott) [reply Makefile.kokkos:1219]: I checked the cmake file and I think it's using the flag there. So I kept it consistent.

Comment (crtrott) [reply Makefile.kokkos:1219]: https://github.com/kokkos/kokkos/blob/54904946f91db5abaa8afd020c47723a4483fb58/cmake/kokkos_arch.cmake#L1287C5-L1287C39

**Comment by stanmoore1** [comment]
I tested this on MI300A and it works as expected for a Makefile build. Thanks @crtrott!

**Review by JBludau — APPROVED** [review]

**Pull request merged**